src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -2777,9 +2788,63 @@ def get_total_BEC(b, CE_index_year, watertap_blocks=None):
             units=CE_index_units,
         )
 
+        # Load the location factor list
+        # User-provide the location as a single (country, city) tuple
+        # Benchmark location: ("United States", "Washington DC")
+        b.location_factor_list = load_location_factor()
+
+        # Build dictionary keyed by (country, city)
+        location_factor_dict = {
+            (entry["country"], entry["city"]): entry["location_factor"]
+            for entry in b.location_factor_list
+        }
+        # Store location factor list for test access
+        b.location_factor_dict = location_factor_dict
+
+        # First tries to match (country, city) exactly.
+        if location in location_factor_dict:
+            location_factor = location_factor_dict[location]["average"]
+        # If no city been provided in location_factors.json, fall back to (country, None)
+        elif (location[0], None) in location_factor_dict:
+            fallback_location = (location[0], None)
+            location_factor = location_factor_dict[fallback_location]["average"]
+        # If country matches but city doesn't, suggest cities
+        elif any(loc[0] == location[0] for loc in location_factor_dict.keys()):
+            # Find all available cities for this country (excluding None)
+            available_cities = sorted(
+                {
+                    loc[1]
+                    for loc in location_factor_dict.keys()
+                    if loc[0] == location[0] and loc[1] is not None
+                }
+            )
+
+            available_city_list = (
+                ", ".join(available_cities) if available_cities else "None"
+            )
+            raise AttributeError(
+                f"No location factor found for {location}. "
+                f"Available cities for '{location[0]}' are: {available_city_list}. "
+                f"Did you mean one of those?"
+            )
+
+        # No country match
+        else:
+            available_countries = sorted(
+                {loc[0] for loc in location_factor_dict.keys()}
+            )
+            raise AttributeError(
+                f"No location factor found for country '{location[0]}'. "
+                f"Available countries are: {', '.join(available_countries)}"
```

Comment:
This is a long list to print to the console. Should we point users to the JSON file instead?

I think for cities it's fine to list it, since each country only has between zero and five or so cities available.